src/lib/utilities/history-summary-plucker.ts

```diff
@@ -0,0 +1,93 @@
+import { findValueIfItIncludesWord } from './find-value-if-it-includes-word';
+
+const summaryExtractors = {
+  ActivityTaskCancelRequested: (d) => ({ Id: d.activityId }),
+  ActivityTaskCompleted: (d) => ({ result: d.result }),
+  ActivityTaskFailed: (d) => ({
+    details: d.details,
+    reason: d.reason,
+  }),
+  ActivityTaskScheduled: (d) => ({
+    'Close Timeout': d.startToCloseTimeout,
+    Id: d.activityId,
+    input: d.input,
+    Name: d.activityType.name,
+  }),
+  ActivityTaskStarted: (d) => ({
+    attempt: d.attempt,
+    identity: d.identity,
+    requestId: d.requestId,
+  }),
+  ActivityTaskTimedOut: (d) => ({
+    'Timeout Type': d.failure?.timeoutFailureInfo?.timeoutType,
+  }),
+  ChildWorkflowExecutionCompleted: (d) => ({
+    result: d.result,
+    Workflow: d,
+  }),
+  ChildWorkflowExecutionStarted: (d) => ({
+    Workflow: d,
+  }),
+  WorkflowTaskCompleted: (d) => ({ identity: d.identity }),
+  WorkflowTaskScheduled: (d) => ({
+    Taskqueue: d.taskQueue.name,
+    Timeout: d.startToCloseTimeout,
+  }),
+  WorkflowTaskStarted: (d) => ({ requestId: d.requestId }),
+  WorkflowTaskTimedOut: (d) => ({ 'Timeout Type': d.timeoutType }),
+  ExternalWorkflowExecutionSignaled: (d) => ({
+    Workflow: d,
+  }),
+  StartChildWorkflowExecutionInitiated: (d) => ({
+    input: d.input,
+    Taskqueue: d.taskQueue.name,
+    Workflow: d.workflowType.name,
+  }),
+  SignalExternalWorkflowExecutionInitiated: (d) => ({
+    input: d.input,
+    signal: d.signalName,
+    Workflow: d,
+  }),
+  TimerStarted: (d) => ({
+    'Fire Timeout': d.startToFireTimeout?.duration,
+    'Timer ID': d.timerId,
+  }),
+  WorkflowExecutionStarted: (d) => {
+    const summary = {
+      'Close Timeout': d.workflowRunTimeout,
+      identity: d.identity,
+      input: d.input,
+      Parent: undefined,
+      Workflow: d.workflowType ? d.workflowType.name : '',
```

Comment:
Prefer `d.workflowType?.name ?? ''`

Reply:
@GiantRobots taught me about this today. I literally didn't know it existed.